### DISCUSSION TITLE
IP Bindings

### DISCUSSION THREAD

**Discussion by Zoey2936** [nextcloud/all-in-one#731] [Ideas] [answered]
Could there maybe be an option to bind the ports of the containers to an IP, beside the manual docker-compose mode? So that the ports Nextcloud talk for example uses could at the same time for another service on another IP-address. (Maybe also the option to use different IPs per Nextcloud aio service/container/port)

**Comment by szaimen** [comment]
What would be the usecase for this?

**Reply by szaimen** [reply]
So basically why would you need port 3478 for another service on the same machine?

**Reply by Zoey2936** [reply]
For example:
1. I would like to bind nextcloud apache itself only to 127.0.0.1, since I don't need it to expose it to the internet, without using a firewall
2. I would like to run big blue button docker on the same machine, which has two public IP-Adresses, but changing in bbb docker the turn / stun ports is very complicated and does not work very well and as I think in Nextcloud aio not supported

**Comment by szaimen** [comment]
Honestly I see only a pretty limited usecase for ip bindings as you are already able to limit access using a firewall.

However allowing to change the talk port does make sense since that is currently not possible and would allow you to run it next to BBB.

**Reply by Zoey2936** [reply]
> However allowing to change the talk port does make sense since that is currently not possible and would allow you to run it next to BBB.

This sounds good


> Honestly I see only a pretty limited usecase for ip bindings as you are already able to limit access using a firewall.

The Problem I have is that I just get my Firewall to block requests to non docker ports, so I can block Ports of non-docker applications, but if I try to block a docker application it seems that docker ignores that rule, but I don't want to remove dockers access to the firewall, but yes I understand that just very few people would need such an option, but Thanks for making it possible to use bbb and nextcloud talk at the sametime using bbb docker and nextcloud aio

**Comment by Zoey2936** [comment]
I've finally got the firewall working to block only port 11000 using:
`/sbin/iptables -I DOCKER-USER -p tcp --destination-port 11000 -i eth0 -j DROP`
And saving this rule

**Reply by Zoey2936** [reply]
Just a small update on this, I changed my command, to:
```sh
# IPv4
iptables -A DOCKER-USER -p tcp -m tcp --dport 11000 -j DROP
# IPv6
ip6tables -A DOCKER-USER -p tcp -m tcp --dport 11000 -j DROP
```

Saving:
```sh
# IPv4
iptables-save > /etc/iptables/rules.v4
# IPv6
ip6tables-save > /etc/iptables/rules.v6
```

Requirements:
iptables & iptables-persistent
I don't know if it works on other OS, but on Debian 11 it works.

**Reply by szaimen** [reply]
Perfect! Thanks for the heads-up! :)

Will add this with https://github.com/nextcloud/all-in-one/issues/884 then!

**Comment by szaimen** [comment, marked as answer]
I will add this with https://github.com/nextcloud/all-in-one/issues/889

**Reply by szaimen** [reply]
This is now released with 1.6.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Reply by Zoey2936** [reply]
I will test it at the end of the week, since I am currently on the baltical sea

**Reply by szaimen** [reply]
All right! Thanks for testing! 
I've planned the stable release for 24.07 or later...

**Reply by Zoey2936** [reply]
> This is now released with 1.6.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

It works

**Reply by Zoey2936** [reply]
Thanks!

**Comment by mamama1** [comment]
sorry for digging this up, but I have a use case. I have a docker host which runs nextcloud and a unifi controller. both want to listen on 3478. Of course, I could just change the talk port (which I'm forced to do now), but I'd rather just bind the nextcloud services to one of the IPs of the docker host and bind the unifi controller to a different IP.

because why not - how hard can it be to expose a IP binding option to a env variable? :)
